### PR TITLE
[prometheus] Add support for renaming Cluster Role and Binding

### PR DESCRIPTION
**Pull request by DannyLyubenov**
#### What this PR does / why we need it
I will be deploying Prometheus in N number of namespaces per cluster and currently `ClusterRole` and `ClusterRoleBinding` get created with the same name as PVC and VolumeMounts which already exists and need to follow specific naming convention

In the interest of not having to decouple the `ClusterRole` and `ClusterRoleBinding` away from the community chart it would be great to be able to rename the 2 objects independently from `fullnameOverride` 

#### Which issue this PR fixes

N/A

#### Special notes for your reviewer

N/A

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: prometheus
-appVersion: v2.43.1
+appVersion: v2.43.2
```
Comment (zeritti) [inline charts/prometheus/Chart.yaml:3]: Please, raise `version`, not `appVersion`. As a new feature is being introduced, the minor version should go up.

File: charts/prometheus/templates/clusterrole.yaml [hunk 2]
```diff
@@ -4,7 +4,11 @@ kind: ClusterRole
 metadata:
   labels:
     {{- include "prometheus.server.labels" . | nindent 4 }}
+{{- if .Values.server.clusterRoleNameOverride }}
```
Comment (zeritti) [inline charts/prometheus/templates/clusterrole.yaml:7]: Depends on the template. Could be as simple as
```
name: {{ include "prometheus.clusterRoleName" . }}
```

File: charts/prometheus/Chart.yaml (hunk 1 above)
Comment (DannyLyubenov) [reply charts/prometheus/Chart.yaml]: apologies for that

File: charts/prometheus/templates/_helpers.tpl [hunk 3]
```diff
@@ -88,6 +88,23 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- end -}}
 {{- end -}}
 
+{{/*
+Create a fully qualified ClusterRole name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "prometheus.clusterRoleNameOverride" -}}
```
[the remark this answers is not in the document]
Comment (DannyLyubenov) [reply charts/prometheus/templates/_helpers.tpl]: Thank you for the proposed idea, I extended it slightly to keep the template cleaner

File: charts/prometheus/templates/_helpers.tpl [hunk 4]
```diff
@@ -88,6 +88,18 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
 
+{{/*
+Create a fully qualified ClusterRole name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "prometheus.clusterRoleName" -}}
+{{- if .Values.server.clusterRoleNameOverride -}}
+{{ .Values.server.clusterRoleNameOverride | trunc 63 | trimSuffix "-" }}
+{{- else -}}
+{{ default (include "prometheus.server.fullname" .) }}
```
Comment (zeritti) [inline charts/prometheus/templates/_helpers.tpl:99]: Thank you. Using the `default` function is unnecessary in this case.

Comment (DannyLyubenov) [reply charts/prometheus/templates/_helpers.tpl]: yes you are right

File: charts/prometheus/Chart.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: prometheus
 appVersion: v2.43.1
-version: 22.4.1
+version: 22.4.2
```
[the remark this answers is not in the document]
Comment (DannyLyubenov) [reply charts/prometheus/Chart.yaml]: done

**Comment by zeritti** [comment]
Thank you, @DannyLyubenov, for your changes. As we are changing the minor version number, will you please drop a line on the new feature in the README's Upgrading section?

**Comment by zeritti** [comment]
Thank you for your contribution, @DannyLyubenov, looks good to me.

**Review by zeritti — APPROVED** [review]

**Pull request merged**